sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClientEncryptionKey.java

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos;
+
+import com.azure.core.util.Context;
+import com.azure.cosmos.implementation.Paths;
+import com.azure.cosmos.models.CosmosClientEncryptionKeyProperties;
+import com.azure.cosmos.models.CosmosClientEncryptionKeyResponse;
+import com.azure.cosmos.models.ModelBridgeInternal;
+import com.azure.cosmos.util.Beta;
+import reactor.core.publisher.Mono;
+
+import static com.azure.core.util.FluxUtil.withContext;
+
+@Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+public class CosmosAsyncClientEncryptionKey {
+    private final CosmosAsyncDatabase database;
+    private String id;
+
+    CosmosAsyncClientEncryptionKey(String id, CosmosAsyncDatabase database) {
+        this.id = id;
+        this.database = database;
+    }
+
+    /**
+     * Get the id of the {@link CosmosAsyncUser}
+     *
+     * @return the id of the {@link CosmosAsyncUser}
+     */
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public String getId() {
+        return id;
+    }
+
+    /**
+     * Set the id of the {@link CosmosAsyncUser}
+     *
+     * @param id the id of the {@link CosmosAsyncUser}
+     * @return the same {@link CosmosAsyncUser} that had the id set
+     */
+    CosmosAsyncClientEncryptionKey setId(String id) {
+        this.id = id;
+        return this;
+    }
+
+    /**
+     * Reads a cosmos client encryption key
+     *
+     * @return a {@link Mono} containing the single resource response with the read client encryption key or an error.
+     */
+    @Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
+    public Mono<CosmosClientEncryptionKeyResponse> read() {
+        return withContext(context -> readInternal(context));
+    }
+
+    private Mono<CosmosClientEncryptionKeyResponse> readInternal(Context context) {
+        String spanName = "readClientEncryptionKey." + getId();
+        Mono<CosmosClientEncryptionKeyResponse> responseMono = this.database.getDocClientWrapper()
+            .readClientEncryptionKey(getLink(), null)
+            .map(response -> ModelBridgeInternal.createCosmosClientEncryptionKeyResponse(response)).single();
+        return database.getClient().getTracerProvider().traceEnabledCosmosResponsePublisher(responseMono, context,
+            spanName,
+            database.getId(),
+            database.getClient().getServiceEndpoint());
+    }
+
+    /**
+     * Replace a cosmos client encryption key
```

Comment:
We should add a note that this method is not meant to be invoked directly.

Reply:
done